src/core/lib/event_engine/poller.h

```diff
@@ -30,20 +30,23 @@ namespace experimental {
 // Work(...).
 class Poller {
  public:
-  // This initial vector size may need to be tuned
-  using Events = absl::InlinedVector<EventEngine::Closure*, 5>;
+  struct Ok {};
   struct DeadlineExceeded {};
   struct Kicked {};
-  using WorkResult = absl::variant<Events, DeadlineExceeded, Kicked>;
+  using WorkResult = absl::variant<Ok, DeadlineExceeded, Kicked>;
 
   virtual ~Poller() = default;
-  // Poll once for events, returning a collection of Closures to be executed.
+  // Poll once for events and process received events. The callback function
+  // is expected to be invoked prior to processing received events.
   //
   // Returns:
-  //  * absl::AbortedError if it was Kicked.
-  //  * absl::DeadlineExceeded if timeout occurred
-  //  * A collection of closures to execute, otherwise
-  virtual WorkResult Work(EventEngine::Duration timeout) = 0;
+  //  * Poller::Kicked if it was Kicked.
+  //  * Poller::DeadlineExceeded if timeout occurred
+  //  * Poller::Ok, otherwise indicating that the callback function was invoked
+  //    and some events were processed.
+  virtual WorkResult Work(
+      EventEngine::Duration timeout,
+      absl::FunctionRef<void()> call_before_processing_events) = 0;
```

Comment:
Let's call the callback something like `poll_again` to make it self-documenting.

Reply:
Done.